metrics/web/test/common/presentation/navigation/metrics_router_delegate_test.dart

```diff
@@ -0,0 +1,153 @@
+      },
+    );
+
+    test(
+      ".currentConfiguration provides the current route configuration of the navigation notifier",
+      () {
+        when(navigationNotifierMock.currentConfiguration).thenReturn(
+          configuration,
+        );
+
+        expect(
+          metricsRouterDelegate.currentConfiguration,
+          equals(configuration),
+        );
+      },
+    );
+
+    test(
+      ".navigatorKey provides the global object key with a value equals to the navigator notifier",
+      () {
+        final key = metricsRouterDelegate.navigatorKey as GlobalObjectKey;
+
+        expect(key.value, isA<NavigationNotifier>());
+      },
+    );
+
+    test(
+      "calls the .notifyListeners() when the navigation notifier calls .notifyListeners()",
+      () {
+        final metricsPageFactory = MetricsPageFactory();
+        final navigationNotifier = NavigationNotifier(metricsPageFactory);
+        final metricsRouterDelegate = MetricsRouterDelegate(navigationNotifier);
+
+        bool isCalled = false;
+        metricsRouterDelegate.addListener(() => isCalled = true);
+
+        navigationNotifier.notifyListeners();
+
+        expect(isCalled, isTrue);
+      },
+    );
+
+    test(
+      ".setInitialRoutePath() delegates call to the .handleInitialRoutePath() method of the navigation notifier",
+      () async {
+        await metricsRouterDelegate.setInitialRoutePath(configuration);
+
+        verify(navigationNotifierMock.handleInitialRoutePath(configuration))
+            .called(equals(1));
+      },
+    );
+
+    test(
+      ".setNewRoutePath() delegates call to the .handleNewRoutePath() method of the navigation notifier",
+      () async {
+        await metricsRouterDelegate.setNewRoutePath(configuration);
+
+        verify(navigationNotifierMock.handleNewRoutePath(configuration))
+            .called(equals(1));
+      },
+    );
+
+    test(
+      ".build() returns the navigator widget",
+      () {
+        const page = MetricsPage(child: Text('test'));
+        final pages = UnmodifiableListView([page]);
+
+        when(navigationNotifierMock.pages).thenReturn(pages);
+
+        final actualWidget = metricsRouterDelegate.build(null);
+
+        expect(actualWidget, isA<Navigator>());
+      },
+    );
+
+    test(
+      "applies the given navigator observers to the navigator widget",
+      () {
+        const page = MetricsPage(child: Text('test'));
+        final pages = UnmodifiableListView([page]);
+
+        final expectedNavigatorObservers = [NavigatorObserver()];
+
+        when(navigationNotifierMock.pages).thenReturn(pages);
+
+        final metricsRouterDelegate = MetricsRouterDelegate(
+          navigationNotifierMock,
+          navigatorObservers: expectedNavigatorObservers,
+        );
+
+        final navigator = metricsRouterDelegate.build(null) as Navigator;
+
+        expect(navigator.observers, equals(expectedNavigatorObservers));
+      },
+    );
+  });
```

Comment:
Should we test that the delegate applies the navigator key to the `Navigator` widget? Also, I believe we should test that this class applies the pages provided by the notifier are applied to the `Navigator` widget.